packages/starlight/components/LanguageSelect.astro

```diff
@@ -46,4 +46,18 @@ function localizedPathname(locale: string | undefined): string {
 		}
 	}
 	customElements.define('starlight-lang-select', StarlightLanguageSelect);
+	window.addEventListener('pageshow', (event) => {
```

Comment:
A better event would be `resume` from the Document API but it's [not available in Safari](https://caniuse.com/mdn-api_document_resume_event) at the moment.